src/Microsoft.OData.Core/UriParser/Visitors/PathSegmentToContextUrlPathTranslator.cs

```diff
@@ -180,6 +180,38 @@ public override string Translate(CountSegment segment)
             return string.Empty;
         }
 
+        /// <summary>
+        /// Translate a FilterSegment
+        /// </summary>
+        /// <param name="segment">the segment to Translate</param>
+        /// <returns>Defined by the implementer.</returns>
+        public override string Translate(FilterSegment segment)
+        {
+            return string.Empty;
+        }
+
+        /// <summary>
+        /// Translate a ReferenceSegment
+        /// </summary>
+        /// <param name="segment">the segment to Translate</param>
+        /// <returns>Defined by the implementer.</returns>
+        public override string Translate(ReferenceSegment segment)
+        {
+            // How a ReferenceSegment should be addressed in the context URI is still a pending decision.
+            // https://github.com/OData/odata.net/issues/1292
+            return string.Empty;

```

Comment:
See comments in 1292.

If the /$ref is in the path, the contextUrl should simply be #$ref (or #Collection($ref).

See [http://docs.oasis-open.org/odata/odata/v4.01/cs01/part1-protocol/odata-v4.01-cs01-part1-protocol.html#sec_EntityReference](http://docs.oasis-open.org/odata/odata/v4.01/cs01/part1-protocol/odata-v4.01-cs01-part1-protocol.html#sec_EntityReference).

Note that the entity set/singleton/navprop is *not* in the Url.  So, for example, Customers(1)/Orders(1)/Items/$ref would just be $metadata#Collection($ref).


That's a bit problemmatic for this translator as encountering a ReferenceSegment should replace the entire ContextUrl fragment with #ref or Collection(#ref).